asynction/server.py

```diff
@@ -113,7 +114,7 @@ def init_app(self, app: Optional[Flask], **kwargs) -> None:
     @classmethod
     def from_spec(
         cls,
-        spec_path: Path,
+        spec_path: Union[Path, AsyncApiSpec],
```

Comment:
```suggestion
        spec_path: Union[Path, JSONMapping],
```

See the previous comment I made with regards to exposing the `AsyncApiSpec` type.

Reply:
@dedoussis Are you saying to just change the type annotations? how would the implementation change? Would I still be able to pass an `AsyncApiSpec` or what would I pass?

Reply:
@alex-zywicki This would also be an implementation change.

I propose the `from_spec` method to accept either a file path or a dict. In the case of dict, the conversion of the given dictionary into an `AsyncApiSpec` object should happen internally within Asynction. This way we can avoid exposing the type to the user. See point 1 of my comment above:

 > Instead of passing the typed object, we could have users passing the raw dict of their AsyncAPI spec. Hence, we would have the from_spec factory method supporting file paths as well as raw dict data. The latter will account for use cases such as deep merging multiple spec files into one dict (the use case of yours) or converting class definitions to JSONSchemata.

Reply:
@dedoussis I guess I'm confused about what you meant when you said a while back

> @alex-zywicki For the time being, If there is a requirement on your end to ensure that each file has a complete and valid AsyncAPI structure, you can use the existing methods of Asynction, but with the risk of having to modify parts of your code when you upgrade to a new minor version of the package.